### PR TITLE
Fix path encoding when working with files

### PR DESCRIPTION
**Pull request by fscholdei**
## Proposed changes

Fix path encoding when working with files

### Your checklist for this pull request

**Contributor**:
- [x] PR is well described and the description can be used as a commit message on squash
- [x] Related issues linked to PR if existing and labels set
- [x] New code is covered with unit tests
- [ ] New ui components are tested inside the storybook (module ui-components only) 
- [x] [Changelog entry file](https://github.com/scm-manager/changelog#changelog-entry-files) created in `gradle/changelog` or CHANGELOG.md is updated for plugins
- [ ] Feature has been tested with different permissions
- [ ] Documentation updated (only necessary for new features or changed behaviour)

**Reviewer**:
- [ ] The clean code principles are respected ([CleanCode](https://clean-code-developer.com/virtues/))
- [ ] All new code/logic is implemented on the right spot / "Should this be done here?"
- [ ] UI changes fits into the layout
- [ ] The UI views / components are responsive (mobile views)
- [ ] Correct translations are available

### Checklist for branch merge request (not required for forks)

- [ ] Branch is green/blue on [Jenkins](https://oss.cloudogu.com/jenkins/)
- [ ] Quality Gate passed on [SonarQube](https://sonarcloud.io/organizations/scm-manager/projects)

### CODE REVIEW

File: src/main/js/Edit/encodeFilePath.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+export function encodeFilePath(path?: string, withEndingSlash?: boolean) {
```
Comment (eheimbuch) [inline src/main/js/Edit/encodeFilePath.ts:24]: How about some unit tests? This code is looks like it is easy to break since it is difficult to understand.

**Review by eheimbuch — COMMENTED** [review]
Changelog?

File: src/main/js/Edit/encodeFilePath.ts (hunk 1 above)
Comment (fscholdei) [reply src/main/js/Edit/encodeFilePath.ts:24]: That's right, that might be really useful.

**Comment by eheimbuch** [comment]
Unfortunately it doesn't work. I can create a new file with `#`. If I go to the "Code" section, I see the file, but if i click on the file I get a blank page, because the link is not encoded properly. Also the filename is decoded on most occurrences, but not on the breadcrumb.
breadcrumd_encoding

**Comment by eheimbuch** [comment]
Link if i click on the file without proper encoding

wrong_link

**Pull request merged**